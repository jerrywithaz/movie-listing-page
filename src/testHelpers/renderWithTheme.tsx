import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from 'containers/ThemeProvider';

function renderWithTheme(component: JSX.Element) {
    return renderer.create(
        <ThemeProvider>
          {component}
        </ThemeProvider>
      );
}

export default renderWithTheme;