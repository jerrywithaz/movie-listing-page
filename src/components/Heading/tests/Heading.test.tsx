import React from 'react';
import { Heading1, Heading2 } from './../';
import renderWithTheme from 'testHelpers/renderWithTheme';

describe("components/Heading", () => {

    describe("Heading 1", () => {

        it("should match snapshot", () => {

            const tree = renderWithTheme(<Heading1>Heading 1</Heading1>).toJSON();
    
            expect(tree).toMatchSnapshot();
    
        });

    });

    describe("Heading 2", () => {

        it("should match snapshot", () => {

            const tree = renderWithTheme(<Heading2>Heading 2</Heading2>).toJSON();
    
            expect(tree).toMatchSnapshot();
    
        });

    });

});