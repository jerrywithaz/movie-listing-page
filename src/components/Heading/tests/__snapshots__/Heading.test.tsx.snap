// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Heading Heading 1 should match snapshot 1`] = `
.c0 {
  font-size: 26px;
  line-height: 30px;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  color: #000000;
  text-transform: uppercase;
  margin: 0px 0px 24px 0px;
}

<h1
  className="c0"
>
  Heading 1
</h1>
`;

exports[`components/Heading Heading 2 should match snapshot 1`] = `
.c0 {
  font-size: 16px;
  line-height: 20px;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: #6B6B6B;
  text-transform: uppercase;
  margin: 24px 0px;
}

<h2
  className="c0"
>
  Heading 2
</h2>
`;
